src/app/Fake.Runtime/SdkAssemblyResolver.fs

```diff
@@ -124,41 +126,105 @@ type SdkAssemblyResolver(logLevel:Trace.VerboseLevel) =
+                | Some varValue -> Some(transformer varValue |> FileInfo)
+                | None -> None)
+
+        let PATHSeparator =
+            if isUnix then
+                ':'
+            else
+                ';'
+
+        let tryFindFromPATH () =
+            Environment
+                .GetEnvironmentVariable("PATH")
+                .Split(PATHSeparator, StringSplitOptions.RemoveEmptyEntries ||| StringSplitOptions.TrimEntries)
+            |> Array.tryPick (fun d ->
+                let fi = Path.Combine(d, dotnetBinaryName) |> FileInfo
+
+                if fi.Exists then
+                    Some fi
+                else
+                    None)
+
+        let  tryFindFromDefaultDirs () =
+            let windowsPath = $"C:\\Program Files\\dotnet\\{dotnetBinaryName}"
+            let macosPath = $"/usr/local/share/dotnet/{dotnetBinaryName}"
+            let linuxPath = $"/usr/share/dotnet/{dotnetBinaryName}"
+
+            let tryFindFile p =
+                let f = FileInfo p
+
+                if f.Exists then
+                    Some f
+                else
+                    None
+
+            if isWindows then
+                tryFindFile windowsPath
+            else if isMac then
+                tryFindFile macosPath
+            else if isLinux then
+                tryFindFile linuxPath
+            else
+                None
+        
+        /// <summary>
+        /// provides the path to the `dotnet` binary running this library, respecting various dotnet <see href="https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables#dotnet_root-dotnet_rootx86%5D">environment variables</see>.
+        /// Also probes the PATH and checks the default installation locations
+        /// </summary>
+        let dotnetRoot =
+            tryFindFromEnvVar () |> Option.orElseWith tryFindFromPATH |> Option.orElseWith tryFindFromDefaultDirs
+        
         let referenceAssembliesPath =
-            dotnetHostPath
-            </> "packs"
-            </> "Microsoft.NETCore.App.Ref"
-            </> this.ResolveSdkRuntimeVersion()
-            </> "ref"
-            </> "net" + this.SdkVersionRaw
-
-        if this.LogLevel.PrintVerbose then
-            Trace.tracefn $"Resolved referenced SDK path: {referenceAssembliesPath}"
-        match Directory.Exists referenceAssembliesPath with
-        | true ->
-            Directory.GetFiles(
-                referenceAssembliesPath,
-                "*.dll"
-            )
-            |> Seq.toList
-        | false ->
-            failwithf "Could not find referenced assemblies in path: '%s', please check installed SDK and runtime versions" referenceAssembliesPath
+            dotnetRoot
+            |> Option.map (fun dotnetRoot ->
+                dotnetRoot.Directory.FullName
+                </> "packs"
+                </> "Microsoft.NETCore.App.Ref"
+                </> this.ResolveSdkRuntimeVersion()
```

Comment:
Not something I want to tackle in this PR but `this.ResolveSdkRuntimeVersion()` is just going to throw if no `global.json` is present. This feels wrong, could the version not be extracted from running `dotnet --version` instead if no global.json is present?

Reply:
agree here, we can't assume the presence of global.json. the safest path is to invoke the `dotnet` found from the new code you added and call `--version`. Proj-Info also does this [here](https://github.com/ionide/proj-info/blob/main/src/Ionide.ProjInfo/Library.fs#L101-L110)

Reply:
Why doesn't `dotnet --version` work for all cases?